backend/webapi/Features/Banners/Index.cs

```diff
@@ -0,0 +1,59 @@
+namespace Pidp.Features.Banners;
+
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+using Pidp.Data;
+
+public class Index
+{
+    public class Query : IQuery<List<Model>>
+    {
+        public string Component { get; set; } = string.Empty;
+    }
+
+    public class Model
+    {
+        public int Id { get; set; }
+        public string Component { get; set; } = string.Empty;
+        public string Status { get; set; } = string.Empty;
+        public string Header { get; set; } = string.Empty;
+        public string Body { get; set; } = string.Empty;
+        public Instant StartTime { get; set; }
+        public Instant EndTime { get; set; }
+
+    }
+
+
```

Comment:
we should add a validator that makes `Component` a required parameter

Reply:
fixed.